### PR TITLE
Progress track to simplify Slider integration

### PR DESCRIPTION
**Pull request by StaehliJ**
## Description

Try to help integrator to implement a Slider with Pillarbox in Compose.

## Changes made

- Add helper for Slider integration `ProgressTracker` that is usable with Compose.

## Checklist

- [ ] Your branch has been rebased onto the `main` branch.
- [ ] APIs have been properly documented (if relevant).
- [ ] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).
- [ ] All pull request status checks pass.

### CODE REVIEW

File: pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt [hunk 1]
```diff
@@ -181,6 +181,15 @@ fun Player.getPlaybackSpeed(): Float {
     return playbackParameters.speed
 }
 
+/**
+ * Current position percent
+ *
+ * @return the current position in percent [0,1].
+ */
+fun Player.currentPositionPercent(): Float {
```
Comment (defagos) [inline pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt:189]: Percentage? Or are they other examples in official APIs where such a method would be called Percent?

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+
+/**
+ * Progress tracker
+ *
+ * Handle a progress position that is a mix of the player current position and the user seeked position.
+ *
+ * @property player The player to track current position.
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt:29]:
```suggestion
 * @property player The player whose current position must be tracked.
```

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+
+/**
+ * Progress tracker
+ *
+ * Handle a progress position that is a mix of the player current position and the user seeked position.
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt:27]:
```suggestion
 * Handle a progress position that is a mix of the player current position and the user desired seek position.
```

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+     * User seek at percent position
+     *
+     * @param percent Position in percent [0,1].
+     */
+    fun userSeek(percent: Float) {
+        userSeekState.value = UserSeekState.Seeking(percent)
+    }
+
+    /**
+     * User has finish seeking.
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt:69]:
```suggestion
     * User has finished seeking.
```

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    fun userSeekFinish() {
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt:71]:
```suggestion
    fun userSeekFinished() {
```

or

```suggestion
    fun finishUserSeek() {
```

maybe?

**Review by defagos — APPROVED** [review]

File: pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt (hunk 1 above)
Comment (StaehliJ) [reply pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt:189]: Yes it is`Player.bufferedPercentage` to get the buffering percentage. It is a int between [0,100|. So I guess we can rename it to `currentPositionPercentage`.

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt (hunk 5 above)
Comment (StaehliJ) [reply pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/ProgressTracker.kt:71]: I have a preference for `userSeekFinished`.

File: pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt (hunk 1 above)
Comment (defagos) [reply pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxPlayer.kt:189]: You should likely rename other occurrences of `percent` as `percentage` then 😇

Have fun! 👺

**Pull request merged**